internal/core/flaw/flaw_controller.go

```diff
@@ -95,22 +108,58 @@ func (c flawHttpController) Read(ctx core.Context) error {
 
 	// get all the associated cwes
 
-	return ctx.JSON(200, flawDTO{
-		ID:                 flaw.ID,
-		ScannerID:          flaw.ScannerID,
-		Message:            flaw.Message,
-		AssetID:            flaw.AssetID.String(),
-		State:              flaw.State,
-		CVE:                flaw.CVE,
-		Component:          flaw.Component,
-		CVEID:              flaw.CVEID,
-		ComponentPurlOrCpe: flaw.ComponentPurlOrCpe,
-		Effort:             flaw.Effort,
-		RiskAssessment:     flaw.RiskAssessment,
-		RawRiskAssessment:  flaw.RawRiskAssessment,
-		Priority:           flaw.Priority,
-		ArbitraryJsonData:  flaw.GetArbitraryJsonData(),
-		LastDetected:       flaw.LastDetected,
-		CreatedAt:          flaw.CreatedAt,
+	return ctx.JSON(200, detailedFlawDTO{
+		flawDTO: flawDTO{
+			ID:                 flaw.ID,
+			ScannerID:          flaw.ScannerID,
+			Message:            flaw.Message,
+			AssetID:            flaw.AssetID.String(),
+			State:              flaw.State,
+			CVE:                flaw.CVE,
+			Component:          flaw.Component,
+			CVEID:              flaw.CVEID,
+			ComponentPurlOrCpe: flaw.ComponentPurlOrCpe,
+			Effort:             flaw.Effort,
+			RiskAssessment:     flaw.RiskAssessment,
+			RawRiskAssessment:  flaw.RawRiskAssessment,
+			Priority:           flaw.Priority,
+			ArbitraryJsonData:  flaw.GetArbitraryJsonData(),
+			LastDetected:       flaw.LastDetected,
+			CreatedAt:          flaw.CreatedAt,
+		},
+		Events: flaw.Events,
 	})
 }
+
+func (c flawHttpController) CreateEvent(ctx core.Context) error {
+
+	flawId, err := core.GetFlawID(ctx)
+	if err != nil {
+		return echo.NewHTTPError(400, "invalid flaw id")
+	}
+
+	flaw, err := c.flawRepository.Read(flawId)
+	if err != nil {
+		return echo.NewHTTPError(404, "could not find flaw")
+	}
+	userID := core.GetSession(ctx).GetUserID()
+
+	var status FlawStatus
+	err = json.NewDecoder(ctx.Request().Body).Decode(&status)
+	if err != nil {
+		return echo.NewHTTPError(400, "invalid payload").WithInternal(err)
+	}
+
+	statusType := status.StatusType
```

Comment:
you need to validate, if the statusType is inside the "allowed" status types. The user can send any string as status type, but we should only accept a small subset of that like: "accepted", "markedForMitigation" ...